bindings/cpp/AndroidSecureStorage/app/src/main/java/com/cronos/play/SecureStorage.kt

```
@@ -0,0 +1,91 @@
+import java.io.FileOutputStream
+import java.io.FileInputStream
+
+
+class SecureStorage {
+
+
+    companion object {
+        init {
+        }
+
+
+
+        @JvmStatic
+        fun readSecureStorage(context:Context ,key:String):   HashMap<String, String> {
+
+            val masterKey: MasterKey = Builder(context)
+                .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                .build()
+
+            val file = File(context.filesDir, key)
+            val encryptedFile: EncryptedFile = EncryptedFile.Builder(
+                context,
+                file,
+                masterKey,
+                EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB
+            ).build()
+
+            val myMap = HashMap<String, String>()
+
+
+            if (file.exists()) {
+                encryptedFile.openFileInput().use { inputStream ->
+                    var myvalue=inputStream.readBytes().toString(Charsets.UTF_8)
+                    myMap["result"] = myvalue
+                    myMap["success"]="1"
+                    myMap["error"]=""
+                }
+            }
+            else {
+                myMap["result"] =""
+                myMap["success"]="0"
+                myMap["error"]="encrypt file not found"
+            }
+
+            return myMap
+        }
+
+        @JvmStatic
+        fun writeSecureStorage(context:Context , key:String, value:String): Int {
+            val masterKey: MasterKey = Builder(context)
+                .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                .build()
+
+            val file = File(context.filesDir, key)
+            val encryptedFile: EncryptedFile = EncryptedFile.Builder(
+                context,
+                file,
+                masterKey,
+                EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB
+            ).build()
+
+
+            var ret=0; // 1: success, 0: fail
+            if (file.exists()) {
+                file.delete()
+            }
```

maybe return an error or have a "force-overwrite" argument? what if someone overwrites it by mistake?

---

rust keyring crate `overwrite` by default, i'll check to prevent overwrite

---

Ok, if that's the keyring default behaviour, we can keep it consistent. Just perhaps in top-level docs, we should mention / warn it'll overwrite